foxglove_data_platform/client.py

```diff
@@ -318,6 +317,42 @@ def get_messages(
             for _, channel, message, decoded_message in reader.iter_decoded_messages()
         ]
 
+    def iter_messages(
```

Comment:
Is `messages` the right name here, if the yielded type includes messages and other types?

Reply:
it's message and message-adjacent data (the schema for the message, the channel for the message). It significantly does not iterate on attachments or metadata. I think the name is decent here.

Reply:
this might be too nitpicky but should the whole method be called `messages`? or is the `iter_` conventional in python or our codebase?

Reply:
I'm partly imagining that once get_messages is deprecated and retired, there may no longer be a need to differentiate these with a prefix